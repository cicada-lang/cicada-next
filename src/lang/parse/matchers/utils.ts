import * as Exps from "../../exp"

export function piBindingtoFnBindingFrom(binding: Exps.PiBinding): Exps.FnBinding {
  switch (binding.kind) {
    case "PiBindingNameless": {
      return Exps.FnBindingName("_")
    }

    case "PiBindingNamed": {
      return Exps.FnBindingName(binding.name)
    }

    case "PiBindingImplicit": {
      return Exps.FnBindingImplicit(binding.name)
    }
  }
}
