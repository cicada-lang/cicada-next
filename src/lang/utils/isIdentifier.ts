const identifierRegExp = new RegExp("^([-_\\p{Letter}][-_\\p{Letter}0-9]*)$", "u")

export function isIdentifier(text: string): boolean {
  return identifierRegExp.test(text)
}
