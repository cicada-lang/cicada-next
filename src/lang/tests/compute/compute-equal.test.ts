import { expect, test } from "vitest"
import { runCode } from "../utils"

test("compute Equal", async () => {
  const output = await runCode(`

compute Equal
compute Equal(String)
compute Equal(String, "abc")
compute Equal(String, "abc", "123")

`)

  expect(output).toMatchSnapshot()
})
