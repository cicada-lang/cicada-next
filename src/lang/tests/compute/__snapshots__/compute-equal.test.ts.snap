// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compute Equal 1`] = `
"(T, from, to) => Equal(T, from, to): (T: Type, T, T) -> Type
(from, to) => Equal(String, from, to): (String, String) -> Type
(to) => Equal(String, "abc", to): (String) -> Type
Equal(String, "abc", "123"): Type"
`;
